server/src/main/java/io/seata/server/storage/redis/lock/RedisLocker.java

```diff
@@ -207,41 +237,22 @@ public boolean isLockable(List<RowLock> rowLocks) {
             List<LockDO> locks = convertToLockDO(rowLocks);
             Set<String> lockKeys = new HashSet<>();
             for (LockDO rowlock : locks) {
-                lockKeys.add(getLockKey(rowlock.getRowKey()));
+                lockKeys.add(buildLockKey(rowlock.getRowKey()));
             }
-            List<String> rowlockJsons = jedis.mget(lockKeys.toArray(new String[0]));
+
             String xid = rowLocks.get(0).getXid();
-            for (String rowlockJson : rowlockJsons) {
-                if (!StringUtils.isEmpty(rowlockJson)) {
-                    LockDO lock = JSON.parseObject(rowlockJson, LockDO.class);
-                    if (lock != null && !Objects.equals(lock.getXid(), xid)) {
-                        return false;
-                    }
-                }
-            }
+            Pipeline pipeline = jedis.pipelined();
+            lockKeys.stream().forEach(key -> pipeline.hget(key, XID));
+            List<String> existedXids = (List<String>) (List) pipeline.syncAndReturnAll();
+            return existedXids.stream().allMatch(existedXid -> xid.equals(existedXid));
```

Comment:
Please pay attention to this problem: list All elements are null
existedXids.size>0

Reply:
finished